src/apps/sketch/communication/HG.js

```diff
@@ -152,8 +152,13 @@ export class Guest {
         console.log({ channel });
         this.channels[channel.label] = channel;
         // channel.onmessage = message => console.log({ channel: message.currentTarget.label, message: message.data });
-        if (Object.keys(this.channels).length >= 4) {
-            this.onConnection()
+        console.log("asf", channel.label)
+        if (channel.label === 'meta') {
+            channel.onmessage = (message) => {
+                if (message.data === 'active_board'){
+                    this.onConnection();
+                }
+            }
```

Comment:
do not use on message channel only for one message, every thing from score to board activity like roles should be handled by meta channel

Reply:
This is for temporary purpose only. I will remove the event listener after getting the message. But when i am working on roles and score, i will move this part there

Reply:
no but we will need meta channel for showing all the participants in the room